engineering-toolkit/eslint-config/src/ecma.js

```diff
@@ -42,7 +42,7 @@ export function ecma(config, ...overrides) {
             },
             name: "import",
             rules: {
-              "import/no-anonymous-default-export": "warn",
+              "import/no-anonymous-default-export": "off",
```

Comment:
Why do we disable it? Default exports are bad pattern in the first place as it allows to have the same thing under unlimited amount of aliases in the code. Having this even if the module author did not export things as default is even worse as it not only inherit the first complain but also allows to write a code that is not aligned with its intentions.

Reply:
But you must import it with the original name and then consciously add an alias. In case of defaults you can make a mistake and you might never notice it.

Reply:
What's wrong with that one?

Reply:
I'm not sure if we have understanding of what this rule actually checks.
This is okay for the rule: `export default function Something() {`
and this is bad: `export default function () {`

But in both cases it's still a default export and you can import it like `import WhatheverNameILike from '...'`.

So IMO this rule just does not make sense - what's the difference if you name something before making a default export or not? Especially that it tends to display false-positives for configuration files.

Can we agree that it makes sense to disable this one or should I revert is as well?

Reply:
Then the question is, why do we want to allow anonymous functions?

Reply:
I think it's just unnecessary requirement - it does not bring anything good imo, just a stylistic change.
On the other hand, we get few warnings about e.g. our [next.config.js](https://github.com/vuestorefront/unified-storefronts/blob/main/apps/storefront-unified-nextjs/next.config.mjs#L29) and boilerplate for i18n: [barrel export function](https://github.com/vuestorefront/unified-storefronts/blob/main/apps/storefront-unified-nextjs/lang/en/index.ts#L4-L6)

IMO these cases are not something I'd like to see in my eslint output 🤷 

FYI: I think it's a minor and I see you gave me a thumbs up, so if I don't get any response to this comment till tomorrow I'll just merge this PR with this rule turned off.

Reply:
okay, leaving this one in the PR